include/rocksdb/table.h

```diff
@@ -462,6 +462,19 @@ struct BlockBasedTableOptions {
   //
   // Default: 256 KB (256 * 1024).
   size_t max_auto_readahead_size = 256 * 1024;
+
+  // If enabled, prepopulate warm/hot data blocks which are already in memory
+  // into block cache at the time of flush. On a flush, the data block that is
+  // in memory (in memtables) get flushed to the device. If using Direct IO,
+  // additional IO is incurred to read this data back into memory again, which
+  // is avoided by enabling this option. This further helps if the workload
+  // exhibits high temporal locality, where most of the reads go to recently
+  // written data.
+  //
+  // Right now, this is enabled only for flush for data blocks. We plan to
+  // expand this option to cover compactions in the future and for other types
+  // of blocks.
+  bool prepopulate_block_cache = false;
```

Comment:
Is there a way to reserve an enum for this? I can see that in the future we might have more choices to prepoluate, only flushing, all SST files, and adaptive. It might be a good idea not to restrict ourselves from future improvements.

Reply:
Sure. I will create an enum for this option.